packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -0,0 +1,288 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BucketOptions, DistributionBucket, DistributionValue, LabelKey, LabelValue, Metric, MetricDescriptor as OCMetricDescriptor, MetricDescriptorType, TimeSeriesPoint, Timestamp} from '@opencensus/core';
+import * as os from 'os';
+import * as path from 'path';
+import {Distribution, LabelDescriptor, MetricDescriptor, MetricKind, Point, TimeSeries, ValueType} from './types';
+
+/**
+ * Util methods to convert OpenCensus Metrics data models to StackDriver
+ * monitoring data models.
+ */
+export class StackdriverStatsExporterUtils {
+  static readonly OPENCENSUS_TASK: string = 'opencensus_task';
+  static readonly OPENCENSUS_TASK_DESCRIPTION: string =
+      'Opencensus task identifier';
+  static readonly OPENCENSUS_TASK_VALUE_DEFAULT =
+      StackdriverStatsExporterUtils.generateDefaultTaskValue();
+
+  /**
+   * Returns a task label value in the format of 'nodejs-<pid>@<hostname>'.
+   */
+  static generateDefaultTaskValue(): string {
+    const pid = process.pid;
+    const hostname = os.hostname() || 'localhost';
+    return 'nodejs-' + pid + '@' + hostname;
+  }
+
+  /**
+   * Converts a OpenCensus MetricDescriptor to a StackDriver MetricDescriptor.
+   * @param metricDescriptor The OpenCensus MetricDescriptor.
+   */
+  static createMetricDescriptorData(
+      metricDescriptor: OCMetricDescriptor, metricPrefix: string,
+      displayNamePrefix: string): MetricDescriptor {
+    return {
+      type: this.getMetricType(metricDescriptor.name, metricPrefix),
+      description: metricDescriptor.description,
+      displayName:
+          this.createDisplayName(metricDescriptor.name, displayNamePrefix),
+      metricKind: this.createMetricKind(metricDescriptor.type),
+      valueType: this.createValueType(metricDescriptor.type),
+      unit: metricDescriptor.unit,
+      labels: this.createLabelDescriptor(metricDescriptor.labelKeys)
+    } as MetricDescriptor;
```

Comment:
Could you remove the `as MetricDescriptor`? The TS compiler should know the object must match the type based on the method return type.